### PR TITLE
Correct OSSL_sleep to support larger times for NonStop PUT model by introducing sleep().

### PR DESCRIPTION
**Pull request by rsbeckerca**
This fix also removes SPT model support as it was previously deprecated. Upcoming threading models on the platform should be supportable without change to this method.

Fixes: #23923
Fixes: #23927
Fixes: #23928

### CODE REVIEW

**Comment by rsbeckerca** [comment]
This should contribute to #23925

File: crypto/sleep.c
Comment (mattcaswell) [inline crypto/sleep.c:36]: This seems identical to the `#else` code below. I wonder if it could be collapsed together? e.g. by making line 25 above

```` C
# elif defined(__TANDEM) && !defined(_REENTRANT)
````

Comment (rsbeckerca) [reply crypto/sleep.c:36]: I'm happy to do that. The last time I submitted this for the original fix, it was recommended to split the #ifdef. Will revise.

Comment (nhorman) [reply crypto/sleep.c:36]: Thank you @rsbeckerca I will approve as soon as you push the requested changes.  Please note that I've marked this as a regression, and our beta freeze is impending on [tuesday, march 26](https://github.com/orgs/openssl/projects/11/views/22).  Given that this will need a 24 hour clock between approval and merge, I would greatly appreciate it if you could get this posted today so we can start that clock running, and make the deadline without the need for additional process gymnastics.

Comment (rsbeckerca) [reply crypto/sleep.c]: I have just pushed the requested fix. This fix is applicable to master and the entire 3.x set of releases.

**Review by nhorman — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Comment by nhorman** [comment]
merged

**Comment by rsbeckerca** [comment]
@nhorman How will we handle 3.2, 3.1, and 3.0? The OSSL_sleep defect is in all branches, but does not show up unless the tests hit that condition.

**Comment by nhorman** [comment]
Well, we don't plan to backport RCU to any older branch (as it's a significant feature), so we're safe regarding this specific test.

That said, I agree it's a general defect that could use backporting, I just didn't want to hold up the 3.3 beta on it.  I would say just open a new issue requesting the backport from the original pr, and label it with openssl-3.2, 3.1 and 3.0.  if you assign it to me I can handle the cherry picks in the next week or so

**Comment by rsbeckerca** [comment]
> Well, we don't plan to backport RCU to any older branch (as it's a significant feature), so we're safe regarding this specific test.
> 
> That said, I agree it's a general defect that could use backporting, I just didn't want to hold up the 3.3 beta on it. I would say just open a new issue requesting the backport from the original pr, and label it with openssl-3.2, 3.1 and 3.0. if you assign it to me I can handle the cherry picks in the next week or so

Thanks. I have created issue #23961 but do not have permission to assign you to it.

**Comment by nhorman** [comment]
I've got it, no worries

**Pull request closed without merging**
